integration_tests/test_search.py

```diff
@@ -323,34 +400,116 @@ class SpecificAuthor(FirstMatch):
+    def __init__(self, author, accept_title=None, threshold=0):
         super(SpecificAuthor, self).__init__(author=author, threshold=threshold)
-        self.accept_book_about_author = accept_book_about_author
+        if accept_title:
+            self.accept_title = accept_title.lower()
+        else:
+            self.accept_title = None
+
+    def author_role(self, expect_author, result):
+        if hasattr(expect_author, 'match'):
+            def match(author):
+                return (
+                    expect_author.search(author.display_name)
+                    or expect_author.search(author.sort_name)
+                )
+        else:
+            expect_author_sort = display_name_to_sort_name(expect_author)
+            expect_author_display = sort_name_to_display_name(expect_author)
+            def match(author):
+                return (
+                    contributor.display_name == expect_author
+                    or contributor.sort_name == expect_author
+                    or contributor.sort_name == expect_author_sort
+                    or contributor.display_name == expect_author_display
+                )
+        for contributor in result.contributors or []:
+            if match(contributor):
+                return contributor.role
+        else:
+            return None
 
     def evaluate_first(self, first):
-        expect_author = self.author
-        author = self._field('author', first)
+        expect = self.original_kwargs['author']
+        if self.author_role(expect, first) is not None:
+            return True
+
         title = self._field('title', first)
-        if expect_author == self.author:
-            return
-        if self.accept_book_about_author and expect_author in title:
-            return
+        subtitle = self._field('subtitle', first)
+        if self.accept_title and (self.accept_title in title or self.accept_title in subtitle):
+            return True
+
         # We have failed.
-        eq_(self.author, author)
+        if hasattr(expect, 'match'):
+            expect = expect.pattern
+        eq_(expect, first.contributors)
 
     def evaluate_hits(self, hits):
-        author = self.author
-        threshold = self.threshold
-        authors = [self._field('author', x) for x in hits]
-        author_matches = [x for x in authors if x == author]
-        self.assert_ratio(author_matches, authors, threshold)
+        last_role = None
+        last_title = None
+        author = self.original_kwargs['author']
+        authors = [hit.contributors for hit in hits]
+        author_matches = []
+        for hit in hits:
+            role = self.author_role(author, hit)
+            author_matches.append(role is not None)
+            last_role = role
+            last_title = hit.title
+        self.assert_ratio(author_matches, authors, self.threshold)
+
+
+class SpecificSeries(Common):
+    """Verify that results come from a certain series of books."""
```

Comment:
We have a lot of tests for books in a series, but we don't have a lot of good .series data. Fortunately the series frequently appears in the title of a book, and we can check that. If necessary we can combine this with an author match to make sure we're not getting unrelated books with the series name in the title.